recipes/linux_installer.rb

```diff
@@ -61,10 +61,26 @@
   mysql_connector_j "#{node['confluence']['install_path']}/lib"
 end
 
-template '/etc/init.d/confluence' do
-  source 'confluence.init.erb'
-  mode '0755'
-  notifies :restart, 'service[confluence]', :delayed
+if node['init_package'] == 'systemd'
+  confluence_service_template = '/etc/systemd/system/confluence.service'
+  confluence_service_template_source = 'confluence.systemd.erb'
+
+  execute 'systemctl-daemon-reload' do
+    command '/bin/systemctl --system daemon-reload'
+    action :nothing
+  end
+else
+  confluence_service_template = '/etc/init.d/confluence'
+  confluence_service_template_source = 'confluence.init.erb'
+end
+
+template confluence_service_template do
+  source confluence_service_template_source
+  owner 'root'
+  group 'root'
+  mode 00755
+  action :create
+  notifies :run, 'execute[systemctl-daemon-reload]', :immediately if node['init_package'] == 'systemd'
```

Comment:
Hm... if `node['init_package'] != 'systemd'` the notification won't be sent at all. 
So, I suggest to make it more explicitly:
```
if node['init_package'] == 'systemd'
  execute 'systemctl-daemon-reload' do
    command '/bin/systemctl --system daemon-reload'
    action :nothing
  end

  template '/etc/systemd/system/confluence.service' do
    source 'confluence.systemd.erb'
    owner 'root'
    group 'root'
    mode 00755
    action :create
    notifies :run, 'execute[systemctl-daemon-reload]', :immediately
  end
else
  template '/etc/init.d/confluence' do
    source 'confluence.init.erb'
    owner 'root'
    group 'root'
    mode 00755
    action :create
    notifies :run, 'execute[systemctl-daemon-reload]'
  end
end
``